docs/web_advanced.rst

```diff
@@ -393,6 +393,82 @@ and safe to use as the key.
 Otherwise, something based on your company name/url would be satisfactory (i.e.
 ``org.company.app``).
 
+
+.. _aiohttp-web-contextvars:
+
+
+ContextVars support
+-------------------
+
+Starting from Python 3.7 asyncio has :mod:`Context Variables <contextvars>` as a
+context-local storage (a generalization of thread-local concept that works with asyncio
+tasks also).
+
+
+*aiohttp* server supports it in the following way:
+
+* A server inherits a current task's context used by the server
+  creation. :func:`aiohttp.web.run_app()` runs a task for handling all underlying dirty
+  job, but :ref:`aiohttp-web-app-runners` can be used as well.
+
+* Application initialization / finalization events (:attr:`Application.cleanup_ctx`,
+  :attr:`Application.on_startup` and :attr:`Application.on_shutdown`,
+  :attr:`Application.on_cleanup`) are executed inside the same context.
+
+  E.g. all context modifications made on application startup a visible on teardown.
+
+* On every request handling *aiohttp* creates a context copy. :term:`web-handler` has
+  all variables installed on initialization stage. But the context modification made by
+  a handler or middleware is invisible to another HTTP request handling call.
+
+An example of context vars usage::
+
+    from contextvars import ContextVar
```

Comment:
Please mark it as a python block

Reply:
Why?
Python is a default language for the project, the doc uses this convention everywhere